src/Parser.jl

```diff
@@ -222,7 +222,7 @@ function parse_object(pc::ParserContext{DictType, <:Real}, ps::ParserState) wher
             # Read value
             value = parse_value(pc, ps)
             chomp_space!(ps)
-            obj[convert(keyT, key)] = value
+            obj[keyT === Symbol ? Symbol(key) : convert(keyT, key)] = value
```

Comment:
This is really icky. Can we leave it and try to come up with a better solution later?

Reply:
What's the advantage to leaving it? This change at least stops everyone receiving a deprecation warning, and doesn't add any new behaviour or remove any existing behaviour.

Reply:
Presumably the deprecation of `String` to `Symbol` conversion was done for a reason, to discourage this kind of implicit conversion. We should rather provide an ability to specify an arbitrary function to process the key (along with the dict type).

Reply:
I think we should just do the easy adaptation to 0.7 now and refactor into something better later.